### DISCUSSION TITLE
IOS: Entire page rendering inside Textbox

### DISCUSSION THREAD

**Discussion by MCAleckson** [unoplatform/uno#8443] [General]
I have an AppBarButton, on page 1, executing Frame.Navigate to page2 containing several TextBoxes.  

After several navigations to page2 and back to page1, a subsequent navigation to page2 will render it with the text boxes filled with the contents of the entire page.  Essentially, it's page2 with all it's textboxes containing full copies of page2.

The attached pictures show page 2 when correctly rendered, and then when the bug occurs.

![pic1](https://user-images.githubusercontent.com/87875638/160466550-656323e9-76b1-49cc-9d31-f78bd79f16fc.png)
![pic2](https://user-images.githubusercontent.com/87875638/160466563-33327030-8652-4108-8f4a-f3175b034733.png)

Thanks for any help you may bring!

**Comment by jeromelaban** [comment]
Thanks for the report. Can you tell which version of uno.ui you are using ?

**Reply by MCAleckson** [reply]
4.0.13

Thanks Jerome!

**Reply by MCAleckson** [reply]
And I just went to uno.ui v4.1.9 - does the same thing...

**Reply by jeromelaban** [reply]
Thanks. We've made changes recently for this specific issue, and may already be present in the 4.2-dev builds. Can you try again with 4.2.0-dev.486 ?

**Reply by MCAleckson** [reply]
I will do so and get back to you.  Thank you!

**Reply by MCAleckson** [reply]
Just installed dev.486, rebuilt the app and the bug is still there.

**Reply by jeromelaban** [reply]
Thanks. Then it's likely a variant of this issue: https://github.com/unoplatform/uno/issues/7610#issuecomment-988090679. Can you try the workaround from that link?

**Reply by MCAleckson** [reply]
Fixed.  Thank you again Jerome!
